index.js

```diff
@@ -32,10 +37,30 @@ const slowDownPlugin = async (fastify, settings) => {
       remaining: remainingRequests
     }
 
+    if (options.skipFailedRequests) {
+      onFinished(reply.raw, err => {
+        if (err || reply.statusCode >= 400) {
+          store.decrementOnKey(key)
+        }
+      })
+    }
+
+    if (options.skipSuccessfulRequests) {
+      onFinished(reply.raw, err => {
+        if (!err && reply.statusCode < 400) {
+          store.decrementOnKey(key)
+        }
+      })
+    }
```

Comment:
are you sure that it's safe to subscribe twice using onFinished?

Reply:
yes, it does support multiple listeners, but I could also do something like this so it only registers one callback:

```js
    if (options.skipFailedRequests || options.skipSuccessfulRequests) {
      onFinished(reply.raw, err => {
        if (options.skipFailedRequests && (err || reply.statusCode >= 400)) {
          store.decrementOnKey(key)
        } else if (
          options.skipSuccessfulRequests &&
          !err &&
          reply.statusCode < 400
        ) {
          store.decrementOnKey(key)
        }
      })
    }
```

I don't know if there's an advantage here since it doesn't make sense for someone to skip both failed and successful requests at the same time and IMO it makes the code less readable.

Reply:
if we know it supports multiple listener I think the code you have already written is more readable